### PR TITLE
Use utilnet.GetIndexedIP instead of replicating the function locally

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Consider that we have implemented `GetIndexedIP` in `k8s.io/utils/net`, it's time to clean up redundant implementations in `k/k`.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: pkg/registry/core/service/ipallocator/allocator.go [hunk 1]
```diff
@@ -187,7 +189,7 @@ func (r *Range) Release(ip net.IP) error {
 // ForEach calls the provided function for each allocated IP.
 func (r *Range) ForEach(fn func(net.IP)) {
 	r.alloc.ForEach(func(offset int) {
-		ip, _ := GetIndexedIP(r.net, offset+1) // +1 because Range doesn't store IP 0
```
Comment (aojea) [inline pkg/registry/core/service/ipallocator/allocator.go]: it also fixed the bug you fixed in the utils repo, right?

Comment (SataQiu) [reply pkg/registry/core/service/ipallocator/allocator.go]: Yes, after that PR gets merged, we can fix the bug by bumping the `utils` version. 😄

Comment (aojea) [reply pkg/registry/core/service/ipallocator/allocator.go]: should we hold this PR until https://github.com/kubernetes/utils/pull/145 merges then?
Also, don't we have to update the vendor folder with the new code too?

Comment (aojea) [reply pkg/registry/core/service/ipallocator/allocator.go]: ahh ok, I got it, we bump `utils` in a follow-up PR so it automatically gets the fix
scratch las comment

**Pull request merged**